xen/include/public/io/sndif.h

```diff
@@ -277,143 +398,157 @@
  * length - read or write data length
- * gref0 - gref9 - references to a grant entries for used pages in read/write
- * request.
- *
- *
- * Request set volume - set/get channels volume in stream:
- *     0    1     2     3     4     5     6     7  octet
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |                      id                       |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |       operation       |       stream_idx      |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |         gref          |       reserved        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |       reserved        |       reserved        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/+
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |       reserved        |       reserved        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
  *
- * id - private guest value, echoed in resp
- * operation - XENSND_OP_SET_VOLUME or XENSND_OP_GET_VOLUME
- * stream_idx - index of the stream (from 0 to 'streams_cnt' - 1.
- *   'streams_cnt' is read from the XenStore)
- * gref - references to a grant entry for page with the volume values
  *
+ * Request set/get volume - set/get channels' volume of the stream given:
+ *          0                 1                  2                3        octet
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                 id                |    operation    |     stream_idx  |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                              reserved                                 |
+ * +-----------------+-----------------+-----------------+-----------------+
+ * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/|
+ * +-----------------+-----------------+-----------------+-----------------+
+ * |                              reserved                                 |
+ * +-----------------+-----------------+-----------------+-----------------+
  *
- * Shared page for set/get volume:
- *
- *     0    1     2     3     4     5     6     7  octet
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |        vol_ch0        |        vol_ch1        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |        vol_ch2        |        vol_ch3        |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * +/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/+
- * +-----+-----+-----+-----+-----+-----+-----+-----+
- * |       vol_ch126       |       vol_ch127       |
- * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * id - private guest value, echoed in resp
+ * operation - XENSND_OP_SET_VOLUME or XENSND_OP_GET_VOLUME
+ * stream_idx - index of the stream ("streams_idx" XenStore entry
+ *   of the stream)
+ * gref[0] is used to exchange volume values from the open request.
```

Comment:
`gref[0]` from `open` reqeuest is used? What if there are audio samples stored right now?
What if multiple `set volume` requests are sent one after another?
What if page on target platform is so small, that it can't hold all 128 volume values?